### PR TITLE
Testing: Use snapshot-diff serializer to remove noise in snapshots

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description
Uses the `snapshot-diff` serializer to remove noise in escaped strings in snapshot tests.

Rebase and merge after #29230 to avoid conflicts.

## How has this been tested?
Snapshot tests run and updated.

## Types of changes
Bug fix (?)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: test/unit/jest.config.js [hunk 1]
```diff
@@ -34,7 +34,11 @@ module.exports = {
 	transform: {
 		'^.+\\.[jt]sx?$': '<rootDir>/test/unit/scripts/babel-transformer.js',
 	},
-	snapshotSerializers: [ 'enzyme-to-json/serializer', 'jest-emotion' ],
+	snapshotSerializers: [
+		'enzyme-to-json/serializer',
+		'jest-emotion',
+		require.resolve( 'snapshot-diff/serializer' ),
```
Comment (gziolo) [inline test/unit/jest.config.js:40]: Out of curiosity, why `require.resolve` call is necessary for this serializer?

File: packages/components/src/ui/scrollable/test/__snapshots__/index.js.snap [hunk 2]
```diff
@@ -78,14 +78,14 @@ exports[`props should render smoothScroll 1`] = `
 - + Second value
```
Comment (gziolo) [inline packages/components/src/ui/scrollable/test/__snapshots__/index.js.snap:78]: Any Idea why this snapshot prints twice:
```
Snapshot Diff:
 - First value
 + Second value
```

There is more outputs like that.

**Review by gziolo — APPROVED** [review]
This is a nice improvement 👍🏻

File: packages/components/src/ui/scrollable/test/__snapshots__/index.js.snap (hunk 2 above)
Comment (sarayourfriend) [reply packages/components/src/ui/scrollable/test/__snapshots__/index.js.snap:78]: Oh that's odd 🤔 I'll do some poking around, thanks for pointing this out.

File: test/unit/jest.config.js (hunk 1 above)
Comment (sarayourfriend) [reply test/unit/jest.config.js]: Whoops, it isn't, I've removed it now.

**Review by gziolo — APPROVED** [review]
It's all beautiful now ❤️ 😄

**Pull request merged**